thamos/cli.py

```diff
@@ -1379,4 +1366,93 @@ def support(output: str) -> None:
             json.dump(info, output_file, sort_keys=True, indent=2)
 
 
+@cli.command("whatprovides")
+@click.pass_context
+@click.option(
+    "--output-format",
+    "-o",
+    type=click.Choice(["json", "yaml", "table"]),
+    default="table",
+    help="Specify output format for the status report.",
+)
+@click.argument("import_name", type=str, required=True)
+@handle_cli_exception
+def whatprovides(import_name: str, output_format: str) -> None:
+    """For a given import_name returns list of (package_name, package_version, index_url) triplets.
+
+    Examples:
+      thamos whatprovides sklearn
+
+    """
+    _LOGGER.info("Returning information on package %r", import_name)
+    result = get_package_from_imported_packages(import_name)
+
+    if output_format == "yaml":
+        yaml.safe_dump({"package_names": result}, sys.stdout)
+    elif output_format == "json":
+        json.dump({"package_names": result}, sys.stdout, indent=2)
+        sys.stdout.write("\n")
+    elif output_format == "table":
+        table = Table()
+
+        header = set()
+        for item in result:
+            for key in item.keys():
+                header.add(key)
+
+        header_sorted = sorted(header)
+        for item in header_sorted:
+            table.add_column(item.capitalize())
+
+        for item in result:
+            row = []
+            for key in header_sorted:
+                entry = item.get(key)
+                row.append(str(entry) if entry is not None else "-")
+
+            table.add_row(*row)
+
+        console = Console()
+        console.print(table, justify="center")
+
+    sys.exit(0)
+
+
+@cli.command("discover")
+@click.pass_context
+@click.option(
+    "--src-path",
+    "-sp",
+    type=str,
+    default=".",
+    envvar="THAMOS_SOURCE_PATH",
+    help="Specify path to consider to discover packages.",
+)
+@click.option(
+    "--runtime-environment",
+    "-r",
+    default=None,
+    metavar="NAME",
+    envvar="THAMOS_RUNTIME_ENVIRONMENT",
+    help="Specify runtime environment to which the given package should be added.",
+)
+@handle_cli_exception
+def discover(runtime_environment: typing.Optional[str], src_path: str = ".") -> None:
+    """Discover packages used in the project.
+
+    Examples:
+      thamos discover
+    """
+    # Obtain list of imports using invectio and verify package from PyPI
+    verified_packages = get_verified_packages_from_static_analysis(src_path=src_path)
+
+    # Update requirements files (Pipfile/Pipfile.lock) or requirements.txt (requirements logic)
+    add_requirements_to_project(
+        requirement=verified_packages,
+        runtime_environment=runtime_environment,
+        index_url="https://pypi.org/simple",
```

Comment:
Can this be taken from results shown on user-api?

Reply:
Done!! Thanks!